src/treemodel/treewalker.js

```diff
@@ -238,13 +250,23 @@ function formatReturnValue( type, item ) {
  * `'CHARACTER'` if walker traversed over a character, or `'TEXT'` if walker traversed over multiple characters (available in
  * character merging mode, see {@link core.treeModel.TreeWalker#constructor}).
  *
- * @typedef {String} core.treeModel.TreeWalkerItemType
+ * @typedef {String} core.treeModel.TreeWalkerValueType
  */
 
 /**
  * Object returned by {@link core.treeModel.TreeWalker} when traversing tree model.
  *
- * @typedef {Object} core.treeModel.TreeWalkerItem
- * @property {treeModel.TreeWalkerItemType} type
- * @property {treeModel.Node|treeModel.TextFragment} item Value between old and new position of {@link core.treeModel.TreeWalker}.
+ * @typedef {Object} core.treeModel.TreeWalkerValue
+ * @property {treeModel.TreeWalkerValueType} type
+ * @property {core.treeModel.Item} item Item between previous and next positions of {@link core.treeModel.TreeWalker}.
```

Comment:
I don't like "between previous and next positions of TreeWalker". Old and new seems more appropriate and doesn't care whether we go `.next()` or `.previous()`. Alternative description might be `Item encountered during TreeWalker iteration` or smth like this.